pandas/core/indexes/timedeltas.py

```diff
@@ -347,24 +347,13 @@ def _partial_td_slice(self, key):
     @Substitution(klass="TimedeltaIndex")
     @Appender(_shared_docs["searchsorted"])
     def searchsorted(self, value, side="left", sorter=None):
-        if isinstance(value, (np.ndarray, Index)):
-            if not type(self._data)._is_recognized_dtype(value):
-                raise TypeError(
-                    "searchsorted requires compatible dtype or scalar, "
-                    f"not {type(value).__name__}"
-                )
-            value = type(self._data)(value)
-            self._data._check_compatible_with(value)
-
-        elif isinstance(value, self._data._recognized_scalars):
-            self._data._check_compatible_with(value)
-            value = self._data._scalar_type(value)
-
-        elif not isinstance(value, TimedeltaArray):
+        if isinstance(value, str):
             raise TypeError(
                 "searchsorted requires compatible dtype or scalar, "
                 f"not {type(value).__name__}"
             )
+        if isinstance(value, Index):
```

Comment:
can likely move this to the base class an use shared impl with Timedelta (and I guess with DTI?)